ipatests/test_integration/test_netgroup.py

```diff
@@ -0,0 +1,149 @@
+# Authors:
+#   Petr Čech <pcech@redhat.com>
+#
+# Copyright (C) 2017  Red Hat
+# see file 'COPYING' for use and warranty information
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import print_function
+
+import pytest
+
+from ipatests.test_integration.base import IntegrationTest
+from ipatests.test_integration.tasks import clear_sssd_cache
+
+
+@pytest.fixture()
+def three_netgroups(request):
+    for i in range(1, 4):
+        request.cls.master.run_command(['ipa', 'user-add',
+                                        'testuser%d' % i,
+                                        '--first', 'Test',
+                                        '--last', 'User%d' % i])
+
+        request.cls.master.run_command(['ipa', 'netgroup-add',
+                                        'test_netgroup%d' % i])
+
+        request.cls.master.run_command(['ipa', 'netgroup-add-member',
+                                        '--users=testuser%d' % i,
+                                        'test_netgroup%d' % i])
+
+    def teardown_three_netgroups():
+        for i in range(1, 4):
+            request.cls.master.run_command(['ipa', 'netgroup-del',
+                                            'test_netgroup%d' % i])
+        for i in range(1, 4):
+            request.cls.master.run_command(['ipa', 'user-del',
+                                            'testuser%d' % i])
+
+    request.addfinalizer(teardown_three_netgroups)
+
+
+class TestNetgroups(IntegrationTest):
+    """
+    Test Netgroups
+    """
+
+    num_clients = 1
+    topology = 'line'
+
+    @classmethod
+    def install(cls, mh):
+        super(TestNetgroups, cls).install(mh)
+
+        cls.client = cls.clients[0]
+        cls.clientname = cls.client.run_command(
```

Comment:
Why do you need a `clientname` AFAIK client should contain hostname as attribute.